### ISSUE TITLE
testfixtures.UseAlterConstraint() -> Does not work with `ON DELETE CASCADE`

### ISSUE THREAD

**Issue by richard-to** [go-testfixtures/testfixtures#67]
Database: Postgres 9.6

I didn't want to make my test user role a superuser, so I tried the `UseAlterConstraint` approach.

When I was testing, I found that one of my tables was always empty, even though it should have been populated with my fixtures.

My guess is that `ON DELETE CASCADE` on my foreign keys is a causing this problem.

Example:

```
parent
 - id

child
  - id
  - parent_id (fkey with on delete cascade)
```

Now this is largely dependent on the order that the data is deleted and inserted. So if child is processed before parent, then the child table will be empty due to the fkey cascade.

```
DELETE FROM "child"
INSERT FIXTURES into "child"
DELETE FROM "parent" (A casade appears to delete from child)
INSERT FIXTURES into "parent"
```

Looking at the codebase, I wonder why the deletes aren't done together. Then after that perform all the necessary inserts? Is there a reason why we need to delete/insert for each table like this?

**Comment by andreynering** [comment]
Hi @richard-to, thanks for the detailed bug report!

There's no particular reason for not clearing all tables together, I just wasn't aware this could cause this kind of problem.

Pull requests are welcome!

**Comment by richard-to** [comment]
Yeah, if I have time I'll see if I can try to tackling this issue. Been busy with other stuff recently.

**Comment by andreynering** [comment]
Fixed as part of #70.
